sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -130,19 +127,12 @@ def __resolve_keyvault_reference(config, key_vault_options, secret_clients):
             secret_clients[key_vault_identifier.vault_url] = referenced_client
 
         if referenced_client:
-            try:
-                return referenced_client.get_secret(
-                    key_vault_identifier.name, version=key_vault_identifier.version
-                ).value
-            except ResourceNotFoundError:
-                raise KeyVaultReferenceError(
-                    "Key Vault %s does not contain secret %s"
-                    % (key_vault_identifier.vault_url, key_vault_identifier.name)
-                )
+            return referenced_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version).value
```

Comment:
Do you mean we want to raise ResourceNotFoundError rather than wrapping it into KeyVaultReferenceError?

Reply:
One of the comments brought up in the review board meeting was that we shouldn't use the customer errors unless we really need one for something. So, yes in this case raise whatever error the SDK throws.